basis/bootstrap/image/primitives/primitives-docs.factor

```diff
@@ -19,7 +19,7 @@ HELP: primitive-quot
 { $description "Creates the defining quotation for the primitive. If 'vm-func' is a string, then it is prefixed with 'primitive_' and a quotation calling that C++ function is generated." } ;
 
 ARTICLE: "bootstrap.image.primitives" "Bootstrap primitives"
-"This vocab contains utilities for declaring primitives to be added to the bootstrap image. It is used by " { $vocab-link "bootstrap.primitives" }
```

Comment:
I think it was (kind of) right before, it's used by the ``core/bootstrap/primitives.factor``.

```
./core/bootstrap/primitives.factor:USING: arrays assocs bootstrap.image.primitives
```

Reply:
`bootstrap.primitives` is not a vocabulary that can be linked to though:
```factor
Factor 0.98 x86.64 (1852, heads/master-e0a93a6ea8, Thu Dec 21 07:25:16 2017)
[Clang (GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final))] on linux
IN: scratchpad USE: bootstrap.primitives
1: USE: bootstrap.primitives
                            ^
Vocabulary does not exist
name "bootstrap.primitives"
```
`help-lint` gives something similar which is why i did that.

`core/bootstrap/primitives.factor` is a file that exists but it must be in error, shouldn't it be called `core/bootstrap/primitives/primitives.factor`?

or maybe the docs meant `bootstrap.image.primitives` (which is a vocab)?

Reply:
So, ``core/bootstrap/primitives.factor`` is a script used in the bootstrap process, not meant to be a vocab you can import, which leads to the error using ``$vocab-link``.  I think ``bootstrap.image.primitives`` is only used by ``core/bootstrap/primitives.factor`` in the entire project.

Reply:
well, can we put the script `core/bootstrap/primitives.factor` in a `primitives` folder so that `help-lint` passes? or should i just ignore the linter?

Reply:
The quickest way is to make it a ``$snippet`` instead of a ``$vocab-link``.

Maybe longer-term we could make ``bootstrap.primitives`` it's own vocab and move the top-level script code into a ``MAIN:`` word.

Reply:
that makes sense, i'll put this in the next set of commits i make on this pull

Reply:
It's also funny that the "script" for ``bootstrap/primitives.factor`` actually does define itself as a vocab when you run it and puts some words in the vocab that it uses in the script part.  Anyway, good cleanup suggestion for later! Thanks!